protocol/communities/manager.go

```diff
@@ -5134,3 +5134,54 @@ func (m *Manager) GetCuratedCommunities() (*CuratedCommunities, error) {
 func (m *Manager) SetCuratedCommunities(communities *CuratedCommunities) error {
 	return m.persistence.SetCuratedCommunities(communities)
 }
+
+func getImage(image *images.IdentityImage) (*common.LinkPreviewThumbnail, error) {
+	thumbnail := &common.LinkPreviewThumbnail{}
+
+	if image.IsEmpty() {
+		return nil, nil
+	}
+
+	width, height, err := images.GetImageDimensions(image.Payload)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get image dimensions: %w", err)
+	}
+
+	dataURI, err := image.GetDataURI()
+	if err != nil {
+		return nil, fmt.Errorf("failed to get data uri: %w", err)
+	}
+
+	thumbnail.Width = width
+	thumbnail.Height = height
+	thumbnail.DataURI = dataURI
+	return thumbnail, nil
+}
+
+func (m *Manager) BuildStatusCommunityLinkPreview(community *Community) (*common.StatusCommunityLinkPreview, error) {
+	communityLinkPreview := &common.StatusCommunityLinkPreview{}
+	if image, ok := community.Images()[images.SmallDimName]; ok {
+		if thumbnail, err := getImage(&images.IdentityImage{Payload: image.Payload}); err != nil {
+			m.logger.Warn("unfurling status link: failed to set community thumbnail", zap.Error(err))
```

Comment:
We access `m.logger` here and probably this keeps us from making this a `Community` method. I think it's fine to drop this logging. It's not much needed anyway.